parlai/agents/transformer/modules.py

```diff
@@ -470,21 +477,34 @@ def forward(self, input, positions=None, segments=None):
 
         tensor *= self.output_scaling
         if self.reduction_type == 'first':
-            return tensor[:, 0, :]
+            output = tensor[:, 0, :]
         elif self.reduction_type == 'max':
-            return tensor.max(dim=1)[0]
+            output = tensor.max(dim=1)[0]
         elif self.reduction_type == 'mean':
             divisor = mask.float().sum(dim=1).unsqueeze(-1).clamp(min=1).type_as(tensor)
             output = tensor.sum(dim=1) / divisor
-            return output
         elif self.reduction_type == 'none' or self.reduction_type is None:
             output = tensor
-            return output, mask
         else:
             raise ValueError(
                 "Can't handle --reduction-type {}".format(self.reduction_type)
             )
 
+        if self.variant in 'bert':
+            output = self.additional_linear_layer(output)
+
+        if (
+            self.reduction_type == 'none'
+            or self.reduction_type is None
+            and self.variant != 'bert'
+        ):
+            # TODO: is the code that passes back the mask still usable? From looking at
```

Comment:
to answer your TODO - yes. E.g. the Polyencoder uses the mask (https://github.com/facebookresearch/ParlAI/blob/master/parlai/agents/transformer/polyencoder.py#L264)

Reply:
Ah, that's great to know - thanks! I've removed the TODO and fixed the conditional. I see that reduction_type == None is not presented as an option in the help string, which makes sense given that using it would cause an error when not using the polyencoder